### DISCUSSION TITLE
How to add "belongs to" associations for composite foreign keys

### DISCUSSION THREAD

**Discussion by MondayCha** [go-gorm/gorm#6935] [Q&A]
Hello, in my project, there is a many-to-many relationship between `User` and `Project`. To manage this, I am using an intermediate table called `UserProject` to associate `User` and `Project`. Here is the structure of `UserProject`:

```go
type UserProject struct {
    gorm.Model
    UserID    uint   `gorm:"primaryKey"`
    ProjectID uint   `gorm:"primaryKey"`
    Role      string `gorm:"type:varchar(32);comment:User's role in the project (admin, user)"`
}
```

Now, users can create tasks in a given project (task belongs to user and project). Here is the design of my Task table:

```go
type AIJob struct {
    gorm.Model
    Name        string      `gorm:"type:varchar(128);not null;comment:Name of the job"`
    UserID      uint
    ProjectID   uint
    UserProject UserProject `gorm:"foreignKey:UserID,ProjectID"`
    TaskType    string      `gorm:"type:varchar(128);not null"`
}
```

However, during migration, it keeps saying that I have not set the appropriate foreign keys. What should I do?

**Comment by utrack** [comment]
Bumping this. Does gorm seriously not support composite FK for one to many?
